src/Microsoft.AspNetCore.SignalR.Core/HubConnectionHandler.cs

```diff
@@ -197,6 +200,8 @@ HubDispatcher<THub> dispatcher
 
                     if (!buffer.IsEmpty)
                     {
+                        connection._receivedMessageThisInterval = true;
```

Comment:
We generally avoid exposing internal flags like this across class boundaries. Let's wrap this up in an `internal` method on `HubConnectionContext` (something like `ResetClientTimeout`)